cmd/openshift-install/gather.go

```diff
@@ -95,30 +96,30 @@ func runGatherBootstrapCmd(directory string) error {
 	if err != nil {
 		if err2, ok := err.(errUnSupportedGatherPlatform); ok {
 			logrus.Error(err2)
-			return unSupportedPlatformGather()
+			return unSupportedPlatformGather(directory)
 		}
 		return errors.Wrapf(err, "failed to get bootstrap and control plane host addresses from %q", tfStateFilePath)
 	}
 
-	logGatherBootstrap(bootstrap, port, masters)
+	logGatherBootstrap(bootstrap, port, masters, directory)
 	return nil
 }
 
-func logGatherBootstrap(bootstrap string, port int, masters []string) {
-	var (
-		sshport string
-		scpport string
-	)
-	if port != 22 {
-		sshport = fmt.Sprintf(" -p %d", port)
-		scpport = fmt.Sprintf(" -P %d", port)
+func logGatherBootstrap(bootstrap string, port int, masters []string, directory string) {
+	logrus.Info("Pulling logs from bootstrap for debugging")
+
+	client, err := ssh.NewClient("core", fmt.Sprintf("%s:%d", bootstrap, port), nil)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	if err := ssh.Run(client, fmt.Sprintf("/usr/local/bin/installer-gather.sh %s", strings.Join(masters, " "))); err != nil {
+		logrus.Fatal(err)
```

Comment:
Do we distinguish between "failed to run the gather at all" and "gather started running, hit some errors, and was only able to create an incommplete tarball for us"?  This fatal error here assumes the former.

Also, I'd be happier with `logGatherBootstrap` returning `error` instead of using `logrus.Fatal` so callers could decide how to compose with any existing errors (e.g. if we're here because an install timed out, the gather error is secondary to the install error).

Reply:
done https://github.com/openshift/installer/pull/1822/commits/cad7f02a8bdfc268ee54ab4f4e25dec1a99640c3